### PR TITLE
Allow disabling of OverrideMethodFilter

### PR DESCRIPTION
**Pull request by glent**

### CODE REVIEW

File: jaxrs/src/main/java/com/proofpoint/jaxrs/JaxrsConfig.java [hunk 1]
```diff
@@ -51,4 +52,18 @@ public boolean isPreload()
     {
         return preload;
     }
+
+    @Deprecated
```
Comment (johngmyers) [inline jaxrs/src/main/java/com/proofpoint/jaxrs/JaxrsConfig.java]: I don't think we should start off deprecating this. We would deprecate after we change the default.

File: jaxrs/src/main/java/com/proofpoint/jaxrs/JaxrsModule.java [hunk 2]
```diff
@@ -196,6 +194,8 @@ public void configure(Binder binder)
             jaxrsBinder(binder).bindAdmin(OverrideMethodFilter.class);
             jaxrsBinder(binder).bindAdmin(ThreadDumpResource.class);
 
+            bindConfig(binder).bind(JaxrsConfig.class);
```
Comment (johngmyers) [inline jaxrs/src/main/java/com/proofpoint/jaxrs/JaxrsModule.java]: I think I'd prefer unconditionally removing `OverrideMethodFilter` from the admin port. There's unlikely to be a use case for it there. Then we wouldn't have to pull the other options into the admin-only module or split the options into two classes.

File: jaxrs/src/main/java/com/proofpoint/jaxrs/OverrideMethodFilter.java [hunk 3]
```diff
@@ -47,22 +50,33 @@
      */
     public static final String METHOD_PARAM = "_method";
 
+    private final boolean filterEnabled;
+
+    @Inject
+    public OverrideMethodFilter(JaxrsConfig config)
+    {
+        requireNonNull(config, "jaxrsConfig is null");
+        filterEnabled = config.isOverrideMethodFilter();
```
Comment (johngmyers) [inline jaxrs/src/main/java/com/proofpoint/jaxrs/OverrideMethodFilter.java]: `JaxrsModule` should use `ConditionalModule.installModuleIf()` to decide whether or not to install this module. The pattern of having a filter that perhaps never does anything is brittle.

File: jaxrs/src/test/java/com/proofpoint/jaxrs/TestOverrideMethodFilter.java [hunk 4]
```diff
@@ -122,9 +122,25 @@ public void testHeaderHasPrecedenceOverQueryParam()
...
+        ContainerRequest request = new ContainerRequest(
+                URI.create("http://www.example.com/"),
+                URI.create("http://www.example.com/?_method=PUT"),
+                "POST",
+                new MockSecurityContext(),
+                new MockPropertiesDelegate());
+        request.header(OverrideMethodFilter.HEADER, "DELETE");
+
+        filter.filter(request);
+        assertEqualsIgnoreCase(request.getMethod(), "POST");
```
Comment (johngmyers) [inline jaxrs/src/test/java/com/proofpoint/jaxrs/TestOverrideMethodFilter.java]: This is testing the implementation, not the API. A better test would be in `TestJaxrsModule` and would `bootstrapTest()` a server with the config property set and then send a request in through its test client.

File: jaxrs/src/main/java/com/proofpoint/jaxrs/JaxrsModule.java [hunk 5]
```diff
@@ -89,7 +92,12 @@ public void configure(Binder binder)
         jaxrsBinder(binder).bind(SmileMapper.class);
         jaxrsBinder(binder).bind(ParsingExceptionMapper.class);
         jaxrsBinder(binder).bind(QueryParamExceptionMapper.class);
-        jaxrsBinder(binder).bind(OverrideMethodFilter.class);
+        newOptionalBinder(binder, Key.get(Ticker.class, JaxrsTicker.class))
+                .setDefault().toInstance(Ticker.systemTicker());
```
Comment (johngmyers) [inline jaxrs/src/main/java/com/proofpoint/jaxrs/JaxrsModule.java]: Why this change?

Comment (glent) [reply jaxrs/src/main/java/com/proofpoint/jaxrs/JaxrsModule.java]: It seems Modules.override doesn't work with AbstractConfigurationAwareModule (got an NPE on buildConfigObject). This change enables the same results (being able to replace the ticker in tests) without requiring a module override. See TestTimingFilter for the related change.

**Comment by johngmyers** [comment]
@glent could you rebase that?
